class-tgm-plugin-activation.php

```diff
@@ -1378,33 +1380,32 @@ protected function _gather_plugin_data() {
 				$i++;
 			}
 
-			// Sort plugins by Required/Recommended type and by alphabetical listing within each type.
-			$resort = array();
-			$req    = array();
-			$rec    = array();
+			return $table_data;
 
-			// Grab all the plugin types.
-			foreach ( $table_data as $plugin ) {
-				$resort[] = $plugin['type'];
-			}
+		}
 
-			// Sort each plugin by type.
-			foreach ( $resort as $type ) {
-				if ( 'Required' === $type ) {
-					$req[] = $type;
-				}
-				else {
-					$rec[] = $type;
-				}
+		/**
+		 * Sort plugins by Required/Recommended type and by alphabetical plugin name within each type.
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param array $items
+		 *
+		 * @return array
+		 */
+		public function sort_table_items( $items ) {
+
+			$type = array();
+			$name = array();
+
+			foreach ( $items as $i => $plugin ) {
+				$type[ $i ] = $plugin['type'];
```

Comment:
Where is this 'type' value originally being added from? I don't see it in example.php, and we don't populate it in `register()`.